### ISSUE TITLE
Make teaser_selection or ContentTeaserProvider configurable for better fallback values

### ISSUE THREAD

**Issue by ampaze** [sulu/sulu#3803]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| Sulu Version | 1.6.14

#### Actual Behavior

When using a **teaser_selection** and a page that has not excerpts, the **title**-field is used as a hardcoded fallback. In ContentTeaserProvider.php:

```php
$title = $document->getField('title')->getValue();
$excerptTitle = $document->getField('excerptTitle')->getValue();
...
('' !== $excerptTitle ? $excerptTitle : $title),
```

#### Expected Behavior

I am using the title, as more of a short name for the page, as that is what is shown in the content treeview and also makes for better URL parts that way. Because of that, the actual headline of the page is a different field than the title field. So I want the headline to be used as a fallback and not title.

#### Possible Solutions

Give teaser_selection more parameters to define the fallbacks (or just the title) 
An alternative way would be to override it globally in ContentTeaserProvider via a a Dependency Injection Parameter.

**Comment by ampaze** [comment]
I would imagine something like

```xml
<property name="pages" type="teaser_selection">
    <meta>
        <title lang="de">Handverlesen</title>
    </meta>
    <params>
        <param name="fallbacks" type="collection">
            <param name="title" value="headline"/>
        </param>
    </params>
</property>
```

**Comment by ampaze** [comment]
Would it be possible to at least make it easier to inherit from ContentTeaserProvider?
Just replace
```php
$title = $document->getField('title')->getValue();
```
with something like
```php
$title = $this->getTitleFromDocument($document);
```
As it stands now, I have to copy a lot of code just to change one line, as well as some private methods used in **ContentTeaserProvider::find**.

**Comment by danrot** [comment]
Fixed in #3884

**Comment by ampaze** [comment]
Awesome!
